profiles-sdk-examples/main-app/src/main/resources/python/submit_job.py

```diff
@@ -112,9 +125,19 @@ def create_from_log_lines(log_parts):
         config_option_overrides = input_params.get("conf")
         if config_option_overrides:
             spark_config.get("pyspark", {}).get("options", {}).get("--conf", {}).update(config_option_overrides)
+        driver_template = get_driver_template("/app/conf/driverTemplate.yaml")
+
+        # variable replace and write new driver podspec
+        # TODO better job of generalizing
+        driver_variables = {
+            'CONTROLLER_NAME': os.environ.get('POD_NAME', "None"),
+            'CONTROLLER_UID': os.environ.get('POD_UID', "None")
+        }
```

Comment:
Do you know what would happen if these variables aren't set?

Reply:
If we are not in a kube cluster then we don't care about these being set so wanted a default.

If we are in a kube cluster and they are not set then we get an error when spark submit attempts to create the driver (when running in cluster mode)